app/assets/admin/js/settings.js

```diff
@@ -37,59 +40,74 @@ jQuery( document ).ready( function ( $ ) {
-				jQuery( '#bp_media_settings_form .bp-media-metabox-holder' ).html( response ).fadeIn( 'slow' );
+				bp_media_metabox_holder.html();
+				bp_media_metabox_holder.html( response ).fadeIn( 'slow' );
 			} );
 		}
 	} );
 
 	/* Cancel Request */
-	jQuery( '#bp-media-settings-boxes' ).on( 'click', '#cancel-request', function () {
+	bp_media_settings_box.on( 'click', '#cancel-request', function () {
 		if ( jQuery( this ).val() ) {
-			jQuery( '#bp_media_settings_form .bp-media-metabox-holder' ).html();
+			bp_media_metabox_holder.html();
 
 			// '<div class="support_form_loader"></div>'
 
-			jQuery( '#bp_media_settings_form .bp-media-metabox-holder' ).html( support_form_loader_div );
+			bp_media_metabox_holder.html( support_form_loader_div );
 			var data = {
 				action: 'rtmedia_cancel_request'
 			};
 
 			// since 2.8 ajaxurl is always defined in the admin header and points to admin-ajax.php
 			jQuery.post( ajaxurl, data, function ( response ) {
-				jQuery( '#bp_media_settings_form .bp-media-metabox-holder' ).html();
-				jQuery( '#bp_media_settings_form .bp-media-metabox-holder' ).html( response ).fadeIn( 'slow' );
+				bp_media_metabox_holder.html();
+				bp_media_metabox_holder.html( response ).fadeIn( 'slow' );
 			} );
 		}
 	} );
 
 	/* Submit Request */
 
-	jQuery( '#bp-media-settings-boxes' ).on( 'submit', '#bp_media_settings_form, .rtmedia-settings-submit', function ( e ) {
+	bp_media_settings_box.on( 'submit', '#bp_media_settings_form, .rtmedia-settings-submit', function ( e ) {
 		var return_code = true;
 		var reg = new RegExp( '^[0-9]+$' );
+		var error_msg = '';
 
-		jQuery( "input[name*='defaultSizes']" ).each( function ( el ) {
-			if ( !reg.test( jQuery( this ).val() ) ) {
-				alert( "Invalid value for " + jQuery( this ).attr( 'name' ).replace( 'rtmedia-options[defaultSizes_', '' ).replace( ']', '' ).replace( /_/g, ' ' ).replace( /(\b)([a-zA-Z] )/g, function ( firstLetter ) {
-					return firstLetter.toUpperCase();
-				} ) );
-				return_code = false;
-				return false;
+		/**
+		 * Validate the default dimensions of the 3 types of media.
+		 * Changed the code from showing alert boxes to appending error messages to the selector.
+		 */
+		$( "input[name*='defaultSizes']" ).each( function () {
+			var current_element = $( this );
+			current_element.css( 'border-color', '#7e8993' ).next( '.error_msg' ).remove();
+			if ( ! reg.test( current_element.val() ) ) {
+				var name_attr = current_element.attr( 'name' );
+				name_attr = name_attr.replace( 'rtmedia-options[defaultSizes_', '' );
+				name_attr = name_attr.replace( ']', '' );
+				name_attr = name_attr.replace( /_/g, ' ' );
+				var error_msg = RTMedia_Admin_Settings_JS.rtmedia_default_sizes_error_message;
+				error_msg = error_msg.replace( '[default_size_property]', name_attr );
+				if( 0 < current_element.parents( '.rtm-content' ).attr( 'id' ).length ) {
+					$( '#tab-' + current_element.parents( '.rtm-content' ).attr( 'id' ) ).click();
```

Comment:
Take 1 variable for `current_element.parents( '.rtm-content' ).attr( 'id' )` and use that to avoid duplicate DOM queries.